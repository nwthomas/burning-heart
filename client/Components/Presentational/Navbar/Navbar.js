import React from "react";
import { StyleSheet } from "react-native";
import styled from "styled-components/native";
import { Link, TouchableOpacity } from "react-router-native";

import searchIcon from "../../../assets/images/search-icon.png";
import homeIcon from "../../../assets/images/home-icon.png";
import profileIcon from "../../../assets/images/profile-icon.png";

const NavbarContainer = styled.View`
  z-index: 10;
  height: 90;
  padding-top: 10;
  align-self: stretch;
  position: absolute;
  flex-direction: row;
  justify-content: space-around;
  left: 0;
  right: 0;
  bottom: 0;
`;

const NavbarImage = styled.Image`
  height: 40;
  width: 40;
  background: #e9eeef;
`;

// Background color for transparent onPress
const linkBackground = "#e9eeef";

const Navbar = _ => {
  return (
    <NavbarContainer>
      <Link underlayColor={linkBackground} to="/search">
        <NavbarImage source={searchIcon} />
      </Link>
      <Link underlayColor={linkBackground} exact to="/">
        <NavbarImage source={homeIcon} />
      </Link>
      <Link underlayColor={linkBackground} to="/profile">
        <NavbarImage source={profileIcon} />
      </Link>
    </NavbarContainer>
  );
};

export default Navbar;
