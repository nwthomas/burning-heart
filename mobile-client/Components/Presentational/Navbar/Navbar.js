import React from "react";
import { StyleSheet, View, Image } from "react-native";
import { Link, TouchableHighlight } from "react-router-native";

import searchIcon from "../../../assets/images/search-icon.png";
import homeIcon from "../../../assets/images/home-icon.png";
import profileIcon from "../../../assets/images/profile-icon.png";

const Navbar = _ => {
  return (
    <View style={styles.container}>
      <Link underlayColor={styles.linkBackground} to="/search">
        <Image style={styles.navbarImg} source={searchIcon} />
      </Link>
      <Link underlayColor={styles.linkBackground} exact to="/">
        <Image style={styles.navbarImg} source={homeIcon} />
      </Link>
      <Link underlayColor={styles.linkBackground} to="/profile">
        <Image style={styles.navbarImg} source={profileIcon} />
      </Link>
    </View>
  );
};

const styles = StyleSheet.create({
  linkBackground: {
    backgroundColor: "#ffffff"
  },
  container: {
    zIndex: 10,
    height: 80,
    paddingTop: 10,
    alignSelf: "stretch",
    position: "absolute",
    flexDirection: "row",
    justifyContent: "space-between",
    paddingLeft: 40,
    paddingRight: 40,
    left: 0,
    right: 0,
    bottom: 0,
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.22,
    shadowRadius: 2.22,
    elevation: 3,
    borderRadius: 8,
    backgroundColor: "#ffffff"
  },
  navbarImg: {
    height: 40,
    width: 40,
    backgroundColor: "#ffffff"
  }
});

export default Navbar;
