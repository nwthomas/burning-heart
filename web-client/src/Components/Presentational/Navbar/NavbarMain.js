import React from "react";

import fireGif from "../../../images/fire.gif";

const NavbarMain = props => {
  return (
    <nav className="navbar__main">
      <div className="navbar__image">
        <img src={fireGif} alt="Burning Heart animated pixel logo" />
      </div>
      <div className="button__container">
        <button className="navbar__button" href="#">
          Login
        </button>
        <button className="navbar__button navbar__button-signup " href="#">
          Sign Up
        </button>
      </div>
    </nav>
  );
};

export default NavbarMain;
