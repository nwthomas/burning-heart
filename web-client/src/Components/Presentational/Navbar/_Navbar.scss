.navbar__container {
  width: 100%;
  background: white;
  height: 80px;
  padding: 0 3%;
  -webkit-box-shadow: 0px 9px 17px -1px rgba(0, 0, 0, 0.18);
  -moz-box-shadow: 0px 9px 17px -1px rgba(0, 0, 0, 0.18);
  box-shadow: 0px 9px 17px -1px rgba(0, 0, 0, 0.18);

  .navbar__main {
    margin: 0 auto;
    width: 100%;
    max-width: 1200px;
    height: 70px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    .burning-heart__name {
      font-size: 3.8rem;
      width: 45%;
      font-weight: 700;
      color: #ffb2b3;

      @media #{$desktop} {
        font-size: 3rem;
      }

      @media #{$mobile} {
        align-items: flex-start;
        width: 70%;
      }
    }

    .navbar__nav-links {
      width: 300px;
      height: 70px;
      display: flex;
      justify-content: flex-end;
      align-items: flex-end;
      padding-bottom: 23px;

      a {
        margin-left: 10%;
        color: black;
        font-weight: 700;

        @media #{$mobile} {
          font-size: 1.2rem;
        }
      }

      a.active {
        text-decoration: underline;
      }
    }
  }
}
