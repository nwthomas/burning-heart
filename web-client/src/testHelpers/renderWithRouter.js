import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default renderWithRouter;
