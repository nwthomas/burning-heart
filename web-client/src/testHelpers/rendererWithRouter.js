import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../store";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Router history={history}>
        <Provider store={store}>{ui}</Provider>
      </Router>
    ),
    history
  };
};

export default rendererWithRouter;
