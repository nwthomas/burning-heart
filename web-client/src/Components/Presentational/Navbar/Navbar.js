import React from "react";

import NavbarMain from "./NavbarMain";

const Navbar = () => {
  return (
    <div className="navbar-container">
      <NavbarMain />
    </div>
  );
};

export default Navbar;
