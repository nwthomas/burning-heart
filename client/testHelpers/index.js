import renderWithRouter from "./renderWithRouter";
import rendererWithRouter from "./rendererWithRouter";

export { renderWithRouter, rendererWithRouter };
