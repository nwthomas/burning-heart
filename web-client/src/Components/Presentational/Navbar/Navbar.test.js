import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../../testHelpers";

import Navbar from "./Navbar";

describe("<Navbar />", () => {
  it("should render the Navbar component without crashing", () => {
    const helpers = renderWithRouter(<Navbar />);
  });

  it("should match the snapshot of Navbar", () => {
    const tree = rendererWithRouter(<Navbar />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
