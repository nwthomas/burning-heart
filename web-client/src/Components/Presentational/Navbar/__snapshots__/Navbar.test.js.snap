// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar /> should match the snapshot of Navbar 1`] = `
<div
  className="navbar__container"
>
  <div
    className="navbar__main"
  >
    <a
      className="burning-heart__name"
      href="/"
      onClick={[Function]}
    >
      Burning Heart
    </a>
    <nav
      className="navbar__nav-links"
    >
      <a
        aria-current="page"
        className="navbar__link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        className="navbar__link"
        href="/home"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        aria-current={null}
        className="navbar__link"
        href="/signup"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </nav>
  </div>
</div>
`;
